src/course-unit/course-xblock/xblock-content/XBlockContent.jsx

```diff
@@ -1,68 +1,153 @@
+import { useRef, useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
-import { useIntl } from '@edx/frontend-platform/i18n';
+import { ensureConfig, getConfig } from '@edx/frontend-platform';
 
 import { LoadingSpinner } from '../../../generic/Loading';
-import AlertMessage from '../../../generic/alert-message';
-import { IFRAME_FEATURE_POLICY } from '../constants';
-import { useIFrameBehavior } from '../hooks';
-import messages from '../messages';
+import { COMPONENT_TYPES } from '../../constants';
+import { blockViewShape, fetchable, IFRAME_FEATURE_POLICY } from '../constants';
+import { wrapBlockHtmlForIFrame } from './iframe-wrapper';
 
+ensureConfig(['STUDIO_BASE_URL', 'SECURE_ORIGIN_XBLOCK_BOOTSTRAP_HTML_URL'], 'studio xblock component');
+
+/**
+ * React component that displays an XBlock in a sandboxed IFrame.
+ *
+ * The IFrame is resized responsively so that it fits the content height.
+ *
+ * We use an IFrame so that the XBlock code, including user-authored HTML,
+ * cannot access things like the user's cookies, nor can it make GET/POST
```

Comment:
I think this comment is not relevant. Maybe to reduce the size of it or just remove.

We can use user's cookies using our workaround.

Reply:
Removed